### PR TITLE
Remove mac binary release artifact.

### PR DESCRIPTION
**Pull request by SpicyLemon**
## Description

This PR updates the release github action. The mac binary will no longer be signed, and it will no longer be part of the release artifacts or upgrade plan.

Reasons for this change:
1. The signing was being done by a account that should no longer be doing so.
2. The Provenance Blockchain Foundation doesn't yet have an Apple Developer ID, and the only reason it would get one is for this signing (there are no other Mac/IOS apps).
3. The mac binary is primarily for use by developers who can hopefully build from source without issues, which doesn't require signing.
4. While I can't say for certain that there aren't any validator nodes running the mac binary, I feel like that's a very good probability. MacOS servers aren't very common, even in the cloud world.
5. Fewer macs are running AMD chipsets these days. Github does not have standard ARM runners though, only AMD. So the compiled mac binary is for older hardware and might not even run on current hardware.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by iramiller — DISMISSED** [review]

**Comment by SpicyLemon** [comment]
Since the `update_release` ("Attach Release Artifacts") job only runs for releases, we won't be able to see it in action until we tag a release. One of the goals of this PR is to no longer include the mac binary as a release artifact, and that is done during this step.

Here's some notes about what's changing and what it means:
* The direct change to this job is to no longer download the mac binary (built in the `build_osx` job), and subsequently, not to upload it as a release artifact. The `build_osx` job will still compile the binary, zip it (with the libwasm library), and upload it as an artifact on the _action_, but it won't be listed as an artifact on the _release_.
* The [`build-release-checksum` build target](https://github.com/provenance-io/provenance/blob/dwedul/no-mac-artifact/Makefile#L271-L277) acts on all `.zip` files in `build/`. By not downloading the mac artifact before running it, it won't be included in the checksum file. That step will run just fine without the mac zip.
* The [`build-release-plan` target/script](https://github.com/provenance-io/provenance/blob/dwedul/no-mac-artifact/scripts/release-plan) is hard-coded to look for the `build/provenance-linux-amd64.zip` and `build/provenance-darwin-amd64.zip` files (downloaded earlier in the job). Since we're not downloading the darwin zip anymore, it won't exist during this step. When the script looks for it, and can't find it, it will just skip it. It therefore won't be included in the release plan.

So, long story short (too late!), the other parts of the release process should still behave as expected.

**Review by egaxhaj — APPROVED** [review]
